docs/aws.md

```diff
@@ -0,0 +1,76 @@
+# AWS Setup Instructions
+The goal of this document is to provide a guide on how to set up the GitHub Action runner on AWS. This document will guide you through the account setup required to use this action with AWS.
+
+## Prerequisites
+- An AWS account
+
+## Setup
+1. Prepare an IAM user with AWS access keys
+    1. Go to the AWS Management Console and sign in to your account.
+    2. Go to the IAM console.
+    3. In the navigation pane, choose Policies.
+    4. Create a new policy by clicking "Create Policy".
+    5. Next, click JSON and paste the following JSON with the required permissions and then click "Next".
+      ```json
+    {
+      "Version": "2012-10-17",
+      "Statement": [
+        {
+          "Effect": "Allow",
+          "Action": [
+            "ec2:RunInstances",
+            "ec2:TerminateInstances",
+            "ec2:DescribeInstances",
+            "ec2:DescribeInstanceStatus"
+          ],
+          "Resource": "*"
+        }
+      ]
+    }
+      ```
+    6. Name the policy (for example: `gha-runner-policy`) and select "Create Policy".
+    7. Now select "Users" in the navigation pane and click "Add user".
+    8. Enter a username (for example: `gha-runner-user`) and ensure that "Provide user access to the AWS Management Console" is unchecked.
+    9. Select "Attach policies directly" and search for the policy you created earlier (`gha-runner-policy`) and select it.
+    10. Then click, "Create user".
+    11. Next, click on the user you just created and go to the "Security credentials" tab.
+    12. Click "Create access key" and then click "Other" and then "Next".
+    13. Click next and then copy the Access Key and Secret Access Key (or download the CSV file), these will not be shown again.
+2. Create your GitHub Access Token
+    1. This can be done with either a Personal Access Token or a Fine-Grained Personal Access Token.
+    2. Go to your GitHub account settings.
+    3. Click on "Developer settings".
+    4. Create a new token with `repo` scope.
+    5. Save and/or copy the token.
+3. Add your credentials to your repository secrets
+    1. Go to your repository on GitHub.
+    2. Click on "Settings", then "Secrets and Variables", and then "Actions".
+    3. Click "New repository secret".
+    4. Add the following secrets:
+      - `AWS_ACCESS_KEY_ID` - The Access Key you copied earlier.
+      - `AWS_SECRET_ACCESS_KEY` - The Secret Access Key you copied earlier.
+      - `GITHUB_TOKEN` - The GitHub token you copied earlier.
+4. Prepare an EC2 image
+    1. Go to the AWS Management Console and sign in to your account.
+    2. Go to the EC2 console.
+    3. Click "Launch Instance".
+    4. Select an Amazon Linux 2 AMI.
+    5. Choose an instance type.
+    6. Configure the instance details.
+    7. Connect to the instance and install `docker`, `git`, and enable the `docker` service. This will be distro dependent.
```

Comment:
probably worth us looking for an AMI that already meets this? Or we could even ship our own AMI (although that's a region-specific hassle).

Reply:
From my understanding, the Deep Learning AMI should work well here but has some limitation in terms of instances that we can ship with it.

Since I am running into my own issues with this (regarding the openmm-gpu-test repo), we might use a tool like [packer](https://www.packer.io/) for building AMIs across cloud providers.

Reply:
With the current pipeline setup for AMIs, I will be able to install CUDA and docker quite easily. Ubuntu comes shipped with `git` so we should be good there. We may need to setup pipelines with Docker as well.